internal/provider/resource_cloudflare_device_posture_integration.go

```diff
@@ -168,6 +173,42 @@ func setDevicePostureIntegrationConfig(integration *cloudflare.DevicePostureInte
 				return fmt.Errorf("api_url is a string")
 			}
 			integration.Config = config
+		case crowdstrike:
+			if config.ClientID, ok = d.Get("config.0.client_id").(string); !ok {
```

Comment:
these error conditions don't really make sense here (looking above, it looks like this was copy pasted but we should still fix it). i think what you're looking to achieve is erroring if the value isn't provided? but then, the schema doesn't enforce it so perhaps we need to be more strict on the checks in here instead?

Reply:
sorry, i think i wasn't clear enough. `return fmt.Errorf("FOO is a string")` isn't a clear error message as to what is happening. let's update these to be specific to the error we're aiming to handle.

Reply:
@suhrit-cf do you need anything else from me here? i'd like to get this out with the next release if we can as this has sat for a bit now.

Reply:
Hi @jacobbednarz made error message more descriptive.